pkg/controller/resourcegraphdefinition/controller_reconcile.go

```diff
@@ -87,9 +92,24 @@ func (r *ResourceGraphDefinitionReconciler) reconcileResourceGraphDefinition(ctx
 }
 
 // setupLabeler creates and merges the required labelers for the resource graph definition
-func (r *ResourceGraphDefinitionReconciler) setupLabeler(rgd *v1alpha1.ResourceGraphDefinition) (metadata.Labeler, error) {
+func (r *ResourceGraphDefinitionReconciler) setupLabeler(
+	rgd *v1alpha1.ResourceGraphDefinition,
+) (metadata.Labeler, metadata.Labeler, error) {
 	rgLabeler := metadata.NewResourceGraphDefinitionLabeler(rgd)
-	return r.metadataLabeler.Merge(rgLabeler)
+	rgSourceLabeler := metadata.NewResourceGraphDefinitionSourceLabeler(rgd)
```

Comment:
is this required to crate applyset for use by instance reconciler? Or are the labels new? Is this needed in this PR?

Reply:
We are introducing a new label. We need this to deal with a bug in KRO where if we have nested RGD (RGD1 creating instance of RGD2), the reconciler of RGD1 and RGD2 both operate on the same label. We added a source label to differentiate it. Similar comments by @jakobmoellerdev that were discussed.